android/src/main/kotlin/com/jhomlala/better_player/BetterPlayerPlugin.kt

```diff
@@ -110,85 +113,113 @@ class BetterPlayerPlugin : FlutterPlugin, ActivityAware, MethodCallHandler {
             if (Build.VERSION_CODES.Q <= Build.VERSION.SDK_INT && Build.VERSION.SDK_INT < Build.VERSION_CODES.S) {
                 if (event == Lifecycle.Event.ON_PAUSE)
                     if (this.showPictureInPictureAutomatically && this.activity?.isInPictureInPictureMode != true) {
-                        this.playerForPictureInPicture?.setupMediaSession(flutterState!!.applicationContext)
                         this.activity?.enterPictureInPictureMode(
                             createPictureInPictureParams(
                                 pipRemoteActions
                             )
                         )
                     }
             }
+            if (event == Lifecycle.Event.ON_DESTROY) {
+                unregisterBroadcastReceiverForPIPAction()
+                _notificationParameter.value = null
+            }
         })
     }
 
-    // To handle action while in picture-in-picture mode.
-    private val broadcastReceiverForPIPAction = object : BroadcastReceiver() {
+    // To handle action while from outside the app.
+    private val broadcastReceiverForExternalAction = object : BroadcastReceiver() {
         // Called when an item is clicked.
         override fun onReceive(context: Context?, intent: Intent?) {
-            if (intent == null || intent.action != DW_NFC_BETTER_PLAYER_CUSTOM_PIP_ACTION) {
+            if (intent == null || intent.action != DW_NFC_BETTER_PLAYER_CUSTOM_ACTION) {
                 return
             }
             when (intent.getIntExtra(EXTRA_ACTION_TYPE, 0)) {
-                PipActions.PLAY.rawValue -> {
-                    playerForPictureInPicture?.tapPlayButtonInPIP()
+                CustomActions.PLAY.rawValue -> {
+                    currentPlayer?.tapExternalPlayButton()
                 }
-                PipActions.PAUSE.rawValue -> {
-                    playerForPictureInPicture?.tapPauseButtonInPIP()
+                CustomActions.PAUSE.rawValue -> {
+                    currentPlayer?.tapExternalPauseButton()
                 }
             }
         }
     }
 
+    private fun registerBroadcastReceiverForExternalAction() {
+        this.activity?.registerReceiver(
+            broadcastReceiverForExternalAction,
+            IntentFilter(DW_NFC_BETTER_PLAYER_CUSTOM_ACTION)
+        )
+    }
+
+    private fun unregisterBroadcastReceiverForPIPAction() {
```

Comment:
Seems this name was not refactored
```suggestion
    private fun unregisterBroadcastReceiverForExternalAction() {
```

Reply:
Thanks, fixed.
https://github.com/dwango-nfc/betterplayer/pull/25/commits/c79a1c7015197d7f44c45f2373c7abc4bde06c06